controllers/subscriber/test/subscriberControllers.test.js

```diff
@@ -0,0 +1,516 @@
+				{ onSubscribe: this._onSubscribe.bind(this), onUnsubscribe: this._onUnsubscribe.bind(this), updateSubscribers: this._updateSubscribers.bind(this) },
+				{}
+			);
+			this._onSubscribeTargets = [];
+			this._onUnsubscribeTargets = [];
+			this._updateSubscribersCalledWith = [];
+		}
+		connectedCallback() {
+			super.connectedCallback();
+			this._eventSubscribers.hostConnected();
+			this._idSubscribers.hostConnected();
+		}
+		disconnectedCallback() {
+			super.disconnectedCallback();
+			this._eventSubscribers.hostDisconnected();
+			this._idSubscribers.hostDisconnected();
+		}
+		getController(controllerId) {
+			if (controllerId === 'event') {
+				return this._eventSubscribers;
+			} else if (controllerId === 'id') {
+				return this._idSubscribers;
+			}
+		}
+		_onSubscribe(target) {
+			this._onSubscribeTargets.push(target);
+		}
+		_onUnsubscribe(target) {
+			this._onUnsubscribeTargets.push(target);
+		}
+		_updateSubscribers(subscribers) {
+			this._updateSubscribersCalledWith.push(subscribers);
+		}
+	}
+);
+
+const combinedRegistry = defineCE(
+	class extends LitElement {
+		constructor() {
+			super();
+			this._subscribers = new SubscriberRegistryController(this,
+				{ onSubscribe: this._onSubscribe.bind(this), onUnsubscribe: this._onUnsubscribe.bind(this), updateSubscribers: this._updateSubscribers.bind(this) },
+				{ eventName: 'd2l-test-subscribe' }
+			);
+			this._onSubscribeTargets = [];
+			this._onUnsubscribeTargets = [];
+			this._updateSubscribersCalledWith = [];
+		}
+		connectedCallback() {
+			super.connectedCallback();
+			this._subscribers.hostConnected();
+		}
+		disconnectedCallback() {
+			super.disconnectedCallback();
+			this._subscribers.hostDisconnected();
+		}
+		getController() {
+			return this._subscribers;
+		}
+		_onSubscribe(target) {
+			this._onSubscribeTargets.push(target);
+		}
+		_onUnsubscribe(target) {
+			this._onUnsubscribeTargets.push(target);
+		}
+		_updateSubscribers(subscribers) {
+			this._updateSubscribersCalledWith.push(subscribers);
+		}
+	}
+);
+
+const eventSubscriber = defineCE(
+	class extends LitElement {
+		constructor() {
+			super();
+			this._subscriberController = new EventSubscriberController(this,
+				{ onError: this._onError.bind(this), onSubscribe: this._onSubscribe.bind(this) },
+				{ eventName: 'd2l-test-subscribe', controllerId: 'event' }
+			);
+			this._onSubscribeRegistry = null;
+			this._onError = false;
+		}
+		connectedCallback() {
+			super.connectedCallback();
+			this._subscriberController.hostConnected();
+		}
+		disconnectedCallback() {
+			super.disconnectedCallback();
+			this._subscriberController.hostDisconnected();
+		}
+		_onError() {
+			this._onError = true;
+		}
+		_onSubscribe(registry) {
+			this._onSubscribeRegistry = registry;
+		}
+	}
+);
+
+const idSubscriber = defineCE(
+	class extends LitElement {
+		static get properties() {
+			return {
+				for: {
+					converter: {
```

Comment:
So here's an example of the converter property to change it from a string of space-delimited ids.  Pretty clean, but means that any component that supports the list (versus just one id) would need to use this for it's `for` property.  Or we can just always expect a string at https://github.com/BrightspaceUI/core/pull/1927/files#diff-647765ef2f64b4feb979d54c882d976462c39916efc668bc36b1e6c85814cbbdR143, and do the string to array parsing there instead. Thoughts?

Reply:
Yeah, or we can just always expect a string and do the array conversion for them in the controller.  That's _probably_ easier...

Reply:
I'm good with that 👍